.vscode/settings.json

```diff
@@ -0,0 +1,15 @@
+{
+  "typescript.tsdk": "node_modules\\typescript\\lib",
+  "files.exclude": {
+    "**/.git": false
+  },
+  "files.watcherExclude": {
+    "**/.git/objects/**": true,
+    "**/.git/subtree-cache/**": true,
+    "**/node_modules/**": true
+  },
+  "javascript.implicitProjectConfig.checkJs": true,
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  }
+}
```

Comment:
Can you please explain the point of each of these configs? And why should we enforce vscode settings on contributors?

Reply:
These settings are some basics that make sure:

- the correct version of typescript is used (rather than the default which is the vscode global version. This can lead to works-on-my-machine type bug reports).
- _if_ the user has the eslint extension enabled, vscode will auto-fix fixable errors on save. It won't do anything for those who don't
- show the `.git` folder in the file explorer. It doesn't do much harm and is occasionally useful to look at.
- tells typescript not to try to watch node_modules and some other folders

Personally I prefer having these settings checked in as it makes for a more consistent developer experiences. Having said that, I realise this can be a bit of a controversial subject, so I've removed this for now - it can be discussed and merged/not-merged separately.

Reply:
Interesting, I like the idea, I think you can open a separate PR with this. Also what does `"javascript.implicitProjectConfig.checkJs": true` do?